setup.py

```diff
@@ -1,25 +1,33 @@
-import sys
+import importlib.util as iutil
 import os
 import platform
+from os.path import dirname, join
+
 import numpy as np
-import pyarrow as pa
-from setuptools import setup, find_packages
-from setuptools import Extension
 from Cython.Distutils import build_ext
+from setuptools import Extension
+from setuptools import setup, find_packages
+
 from aequilibrae.paths.__version__ import release_version
 
-sys.dont_write_bytecode = True
+spec = iutil.find_spec("pyarrow")
+
+include_dirs = [np.get_include()]
+if spec is not None:
+    import pyarrow as pa
 
-here = os.path.dirname(os.path.realpath(__file__))
-whole_path = os.path.join(here, "aequilibrae/paths", "AoN.pyx")
-ext_module = Extension(
-    "aequilibrae.paths.AoN", [whole_path], include_dirs=[np.get_include(), pa.get_include()], language="c++"
-)
+    include_dirs.append(pa.get_include())
+
+whole_path = join(dirname(os.path.realpath(__file__)), "aequilibrae/paths", "AoN.pyx")
+ext_module = Extension("aequilibrae.paths.AoN", [whole_path], include_dirs=include_dirs, language="c++")
```

Comment:
What happens if you build AoN without pyarrow in teh includes list?

Reply:
It crashes (in a rather ugly way) when we try to use with it